code-challenges.js

```diff
@@ -9,67 +9,248 @@
+
+
 var refactorTester1 = true
 var refactorTester2 = 42
 var refactorTester3 = "IAmACodingMaster"
 
+const vowelRemover = (i) =>{
+    //let newVar = [i]
+    let splitString = []
+    if (typeof i  !== "string")
+    return `${i} is not a string`
+    else if(typeof i  === "string")
+    splitString.push(i)
+    return splitString.filter (value =>{
+        
+    return value !== "a" && value !=="e" && value !== "i" && value !== "o" && value !== "u" && value !== "I"
+    // })
+     }).join()
+}
+
+
+console.log(vowelRemover(refactorTester1))
+console.log(vowelRemover(refactorTester2))
+console.log(vowelRemover(refactorTester3))
 
 
 
 
 // --------------------6) Create a function that takes in an array of strings and returns the string with the most characters.
 // Expected output: "chopsticks" "stemware"
 
+
+// setup finction that takes an array counts the number of index spaces
+//retruns the highest number
+
+//set up function that takes in array of strings as argument
+//function must log the index 
+//function must compair index values
+//retruns string with largest index
+
+
 var utensils = ["fork", "knife", "tongs", "chopsticks", "skewer"]
 var vessels = ["cup", "mug", "stemware", "glass", "tumbler", "tea cup"]
 
+// let largestIdex = 
+
+// for (let i =0 ; i < array.length ; i++){
+
+// }
+
+// console.log(largestIdex(utensils))
+
+
+// // const largestIdex = (array) => {
+// //     return 
+    
+    
+// //     array.filter( value, index => {
+// //     })
+// // }
+
+ const largestIndex = (array) => {
+    //  array.forEach(element){  
+    //      if (typeof element !== "string"
+    //      retrun `${newCont.push()}`
+     //}
+ 
+     let highestIndex = 9;
+     let lastIndex = array[i];
+     let newCont = []
+    for ( var i= 1 ; i < array.length ; i++){
+    
+
+    if  (array [i] > highestIndex) {
+        highestIndex = array[i];
+         lastIndex = i;
+
+        }
+        return newCont.push()
+//    finalIndex.push(array.indexOf(3)) ; 
+// }return finalIndex
+
+    } return lastIndex 
+
+}
+console.log(largestIndex(utensils));
+console.log(largestIndex(vessels));
+
 
 
 
 
+
+// console.log(largestIdex(utensils))
+
 // --------------------7) Create a function that takes in an array of strings and returns an array with the last letter removed from each string.
 // Expected output: ["Luig", "Peac", "Toa", "Yosh"] ["Ink", "Blink", "Pink", "Clyd"]
 
 var mario = ["Luigi", "Peach", "Toad", "Yoshi"]
 var pacman = ["Inky", "Blinky", "Pinky", "Clyde"]
+
+
+//setup a finction that takes in a array as an argument 
+//retruns a new array with the last letter removed from each element 
+//using the .pop()
+//using the .map() to create the new array retruning the items minus the last item
+//creating a container for each item and then joining them at the end
+
+//??? i left this one commented out becuase i could not get it to work and I no matter what I tried or looked on line I could not get make it do what i wanted. 
```

Comment:
sure thing! your pseudocode is getting closer. Something that I think might be helpful is if you wrote instructions for what should happen for one value in the array and then try it with the next value in the array.